README.md

```diff
@@ -1,2 +1,560 @@
-# llrp-device-service
-Device Service for LLRP Messages from 3rd Party RFID Readers
+# Device RFID LLRP Go
+EdgeX device service for communicating with LLRP-based RFID readers.
+
+## Table of contents
+
+* [First Run](#first-run)
+* [Device Discovery](#device-discovery)
+* [Device Profiles, Custom LLRP Messages, and Service Limitations](#device-profiles-custom-llrp-messages-and-service-limitations)
+* [Connection Management](#connection-management)
+* [Example Scripts](#example-scripts)
+* [Testing](#testing)
+* [Snap Development and Testing](#snap-development-and-testing)
+* [Footnotes](#footnotes)
+
+## First Run
+**Build Native**
+```bash
+make build
+```
+**Build Docker**
+```bash
+make docker
+```
+**Configure subnet information**
+```bash
+./bin/auto-configure.sh
+```
+**Trigger a device discovery**
+
+*Note: make sure your LLRP devices are plugged in and powered on before this step*
+```bash
+curl -X POST http://localhost:51992/api/v1/discovery
+```
+
+At this point the `device-rfid-llrp` service should have discovered your LLRP devices on the network and
+registered them with EdgeX.
+
+For more detailed info, see [Device Discovery](#Device-Discovery) and 
+[EdgeX Device Naming](#EdgeX-Device-Naming).
+
+## Device Discovery
+*Note: Device discovery is currently only compatible with IPv4 networks.
+If using an IPv6-only network, you will need to [manually add your devices to EdgeX](#manually-adding-a-device).*
+
+This service has the functionality to probe the local network 
+in an effort to discover devices that support LLRP.
+
+This discovery also happens at a regular interval and can be configured via 
```

Comment:
The reason behind the manual call to discover devices is because the `./bin/auto-configure.sh` sets the subnet to search for inside of consul. At that point the service is notified of the configuration change, but it does not trigger a discovery. If the subnet information is already provided via TOML for instance, it would be picked up on startup and ran accordingly.

Timer logic used to not run on startup, but has since been addressed:
[Device SDK Go Issue #547](https://github.com/edgexfoundry/device-sdk-go/issues/547)